dali/operators/image/distortion/jpeg_compression_distortion_op_gpu.cu

```diff
@@ -38,17 +37,63 @@ class JpegCompressionDistortionGPU : public JpegCompressionDistortion<GPUBackend
   std::vector<int> quality_;
 };
 
+template <int ndim>
+TensorListShape<ndim - 1> unfold_dim(const TensorListShape<ndim> &shapes) {
+  static_assert(ndim > 1,
+                "Can't reduce dimentionality of dynamic, or single dimentional TensorListShape");
```

Comment:
I think that doing this on dynamic shapes should be doable. Same with ndim == 1 (would produce scalars)

Reply:
Yes we can, but didn't needed it here and would be more complex.

Reply:
Sure, I meant it in case we make this a shared utility

Reply:
Do we know any other potential users of such transformation?

Reply:
Yeah, I think it could be used quite extensively.